src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/RestartPipelineTest.java

```diff
@@ -0,0 +1,253 @@
+                CONTAINER_ENV_VAR_VALUE);
+        TemplateEnvVar containerSecretEnvVariable = new SecretEnvVar("CONTAINER_ENV_VAR_FROM_SECRET",
+                "container-secret", SECRET_KEY);
+        podTemplate.getContainers().get(0)
+                .setEnvVars(asList(containerEnvVariable, containerEnvVariableLegacy, containerSecretEnvVariable));
+    }
+
+    private PodTemplate buildBusyboxTemplate(String label) {
+        // Create a busybox template
+        PodTemplate podTemplate = new PodTemplate();
+        podTemplate.setLabel(label);
+
+        ContainerTemplate containerTemplate = new ContainerTemplate("busybox", "busybox", "cat", "");
+        containerTemplate.setTtyEnabled(true);
+        podTemplate.getContainers().add(containerTemplate);
+        setEnvVariables(podTemplate);
+        return podTemplate;
+    }
+
+    public void configureCloud() throws Exception {
+        cloud = setupCloud();
+        createSecret(cloud.connect());
+        cloud.getTemplates().clear();
+        cloud.addTemplate(buildBusyboxTemplate("busybox"));
+
+        // Slaves running in Kubernetes (minikube) need to connect to this server, so localhost does not work
+        URL url = story.j.getURL();
+
+        String hostAddress = System.getProperty("jenkins.host.address");
+        if (hostAddress == null) {
+            hostAddress = InetAddress.getLocalHost().getHostAddress();
+        }
+        URL nonLocalhostUrl = new URL(url.getProtocol(), hostAddress, url.getPort(),
+                url.getFile());
+        JenkinsLocationConfiguration.get().setUrl(nonLocalhostUrl.toString());
+
+        story.j.jenkins.clouds.add(cloud);
+    }
+
+    protected String loadPipelineScript(String name) {
+        try {
+            return new String(IOUtils.toByteArray(getClass().getResourceAsStream(name)));
+        } catch (Throwable t) {
+            throw new RuntimeException("Could not read resource:[" + name + "].");
+        }
+    }
+
+    @Test
+    public void runInPodWithRestartWithMultipleContainerCalls() throws Exception {
+        story.addStep(new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+
+                configureCloud();
+
+                story.j.jenkins.addNode(new DumbSlave("slave", "dummy", tmp.newFolder("remoteFS").getPath(), "1",
+                        Node.Mode.NORMAL, "", new JNLPLauncher(), RetentionStrategy.NOOP,
+                        Collections.<NodeProperty<?>>emptyList())); // TODO JENKINS-26398 clumsy
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "p");
+                p.setDefinition(new CpsFlowDefinition(loadPipelineScript("runInPodWithRestartWithMultipleContainerCalls.groovy")
+                        , true));
+                WorkflowRun b = p.scheduleBuild2(0).waitForStart();
+                SemaphoreStep.waitForStart("wait/1", b);
+                SemaphoreStep.success("wait/1", false);
```

Comment:
Makes little sense to wait for a `semaphore` and then immediately let it proceed anyway. What you are actually waiting for is something else, so just drop the `semaphore`.